src/libraries/Common/src/System/Obsoletions.cs

```diff
@@ -89,5 +89,8 @@ internal static class Obsoletions
 
         internal const string X509CertificateImmutableMessage = "X509Certificate and X509Certificate2 are immutable. Use the appropriate constructor to create a new certificate.";
         internal const string X509CertificateImmutableDiagId = "SYSLIB0026";
+
+        internal const string KeyPropertiesMessage = "Key and PrivateKey are obsolete. Use the appropriate method to get the public key, or create a new instance with a different private key.";
```

Comment:
The message feels a little... off.

```suggestion
        internal const string KeyPropertiesMessage = "PublicKey.Key and X509Certificate2.PrivateKey are obsolete. Use the appropriate method to get the key (such as GetRSAPublicKey or GetRSAPrivateKey), or use the CopyWithPrivateKey method to create a new instance with a private key.";
```

Reply:
> use the CopyWithPrivateKey method to create a new instance with a private key.

Well, `PublicKey` has no such method which is why mine was probably too-vague. I'm starting to wonder if they would be better off as two separate diagnostics.

Reply:
Or maybe I am thinking too hard about this.

Reply:
Yeah, I was looking for precedent of using the same ID with different messages, and don't see that.

Numbers are cheap, splitting them so the message is more actionable makes sense to me.   @terrajobst ?

Reply:
I split the diagnostics and think that looks better. I'm happy to revert that last commit and combine then if we come up with how to message both with a single diagnostic message.

Reply:
The way to think about this is like this:

Having two messages for the same diagnostic is fine, but I'd be careful that you don't end up with a grab bag for the ID where someone would want to *frequently* suppress one but not the other. Now, of course there is trade off. The entire point of having a diagnostic ID is to group multiple different APIs under the same ID. You can always *construct* a case where someone would want to suppress it in the context of one API but not the other, but for the imperative word is *frequently*. People can always suppress specific occurrences by using localized `#pragma` suppressions, so rare cases where this is needed always have an escape hatch.

Does this make sense?

Reply:
@terrajobst that is very insightful, thank you. I'm not sure that feedback pushes me over the edge in to re-combining them. Those APIs feel disparate that two separate codes make some sense.

@bartonjs et al. I am happy to change though if anyone thinks otherwise.